sstables_loader.cc

```diff
@@ -494,7 +509,12 @@ future<> sstables_loader::download_task_impl::run() {
     auto [ table_id, sstables_on_shards ] = co_await replica::distributed_loader::get_sstables_from_object_store(_loader.local()._db, _ks, _cf, _sstables, _endpoint, _bucket, _prefix, cfg);
     llog.debug("Streaming sstables from {}({}/{})", _endpoint, _bucket, _prefix);
     co_await _loader.invoke_on_all([this, &sstables_on_shards, table_id] (sstables_loader& loader) mutable -> future<> {
-        co_await loader.load_and_stream(_ks, _cf, table_id, std::move(sstables_on_shards[this_shard_id()]), false, false);
+        co_await loader.load_and_stream(_ks, _cf, table_id, std::move(sstables_on_shards[this_shard_id()]), false, false,
+                                        [this] (unsigned num_streamed) {
+            return smp::submit_to(0, [this, num_streamed] {
```

Comment:
Isn't it a bit heavyweight just for the sake of progress collection?
What if there's a vector of integers, sized at smp::count elements, then each shard ++'s its countera and download_task_impl::get_progress() accumulates it over the whole vector?

Reply:
I agree with the heaviness comment, I don't know how to do it differently, though.
Do we have examples from other features that we could imitate?